### ISSUE TITLE
Suport cuDNN v6.0

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#8828]
cuDNN v6.0 has been released. There are some cool new features:
* Dilated Convolutions: Dilated Convolutions are now supported in cuDNN without a
change in API.
* `cudnnConvolutionBiasActivationForward` allows for the execution of a single kernel fusing convolution, bias and activation operations

Full release notes:
>Dilated Convolutions: Dilated Convolutions are now supported in cuDNN without a
change in API. Previously unused “upscale” fields in the Convolution Descriptor
have been repurposed to allow user specification of dilation factors along each
dimension. Support for dilation is present in the following code paths :
Forward : CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM,
Backward Data: CUDNN_CONVOLUTION_BWD_DATA_ALGO_0 and
Backward Filter: CUDNN_CONVOLUTION_BWD_FILTER_ALGO_0
 The new API cudnnConvolutionBiasActivationForward allows for the execution of
a single kernel fusing convolution, bias and activation operations. At present, only
per channel bias and RELU activation are supported.
 Inference on 8 bit integer data is now supported, leveraging the 4 element dot
product instruction (IDP4A) of Pascal GPUs with CUDA capabilities 6.1. Two tensor
layouts are supported for this feature: CUDNN_TENSOR_NHWC with INT8 data
type and CUDNN_TENSOR_NCHW_VECT with INT8x4 data type.
 RNN now supports 3 algorithms
o CUDNN_RNN_ALGO_STANDARD :
 Same functionality as in CUDNN v5.1
o CUDNN_RNN_ALGO_PERSIST_STATIC :
 This algorithm relies on the usage of persistent CUDA kernels which
are pre-compiled to fit different GPUs.
 This algorithm is available only on Pascal GPUs.
o CUDNN_RNN_ALGO_PERSIST_DYNAMIC :
This algorithm also relies on the usage of persistent CUDA kernels
but these kernels are compiled at runtime using nvrtc. In some cases
this results in a significant performance benefit.
 This algorithm is also available only on Pascal GPUs and is supported
only on Linux and Windows.
 Support for 1D-FFT convolutions has been added
 New API routine cudnnReduceTensor has been added, supporting 8 reduction
operations
 Activation mode CUDNN_ACTIVATION_ELU is now supported.
 A deterministic max pooling mode CUDNN_POOLING_MAX_DETERMINISTIC
has been added.
 Significant performance improvement for softmax layers for mode
CUDNN_SOFTMAX_MODE_CHANNEL has been achieved when low batch
number is used.
 Significant performance improvements have been added for cudnnAddTensor
when spatial dimensions are set to 1.

**Comment by gunan** [comment]
We base our builds off of nvidia/cuda docker images: https://hub.docker.com/r/nvidia/cuda/
Once cudnn 6 appears here, we will try to update our builds very quickly.

**Comment by cancan101** [comment]
@gunan not sure if you were referring to this, but part of this ticket is about not just new docker images, but also updating the TF kernels to take advantage of the new features in cuDNN 6. For example, [pyTorch uses cuDNN 6 for dilated convolutions](https://github.com/pytorch/pytorch/blob/8aa1cefed880e3ad1a839bdbdc9d1fb9d52f5af3/torch/nn/_functions/conv.py#L81-L82).

**Comment by gunan** [comment]
I understand, however moving our build infra to cudnn6 will have to be the first step.
For updating ops, and a timeline for that, @zheng-xq should comment.

**Comment by flx42** [comment]
Please note that the new images will probably break your build because of #8264, see the details on GitLab:
https://gitlab.com/nvidia/cuda/issues/2#note_26563587

**Comment by cancan101** [comment]
okay, @gunan https://hub.docker.com/r/nvidia/cuda/ now has cudnn6 images.

**Comment by b52** [comment]
Is there an ETA for this issue?

**Comment by gnujimmik** [comment]
Hello, 

About the cudnn 6.0 compatibility, is it already working? I am trying tensorflow installation from source and I have got the error:

> Please specify the location where cuDNN  library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]: 
Invalid path to cuDNN 6.0.21 toolkit. Neither of the following two files can be found:
/usr/local/cuda-8.0/lib64/libcudnn.so
/usr/local/cuda-8.0/libcudnn.so

The OS is ubuntu 16.04, CUDA 8.0 and used anaconda for python 2.7

**Comment by Androbin** [comment]
@gnujimmik 
Seems like you have installed CUDA but not cuDNN.
Please refer to: [https://developer.nvidia.com/cudnn](https://developer.nvidia.com/cudnn)

**Comment by flx42** [comment]
@Androbin @gnujimmik actually, `libcudnn.so` might be in the standard install path:
```
/usr/lib/x86_64-linux-gnu/libcudnn.so
```

**Comment by Androbin** [comment]
@flx42 
As far as I know, cuDNN isn't shipped with CUDA and has to be dropped into the directory separately.

**Comment by flx42** [comment]
@Androbin it's not shipped with CUDA indeed, but you can install cuDNN in two ways on Ubuntu:
1. Extract an archive anywhere in your filesystem.
2. Use our machine learning deb package repository and `apt-get install -y libcudnn6`

On CentOS, you only have option 1.

**Comment by gnujimmik** [comment]
Actually I am making use of a docker container from nvidia/cuda:8.0-cudnn6-runtime-ubuntu16.04 
My server already has cuda 8.0 and cudnn 6 and I can see it from /usr/local/cuda-8.0 from the server, but not in the docker container. Will it be an issue of the docker image?

**Comment by danielbaak** [comment]
@gnujimmik have you tried cudnn 6.0 with tensorflow 1.2-rc0 ?

**Comment by gnujimmik** [comment]
@danielbaak yes. ubuntu 16.04 + cuda 8.0 + cudnn 6.0 + tensorflow 1.2-rc0.

**Comment by qieaaa** [comment]
@gnujimmik hi,what about cuDNN6.0  with tensorflow 1.1? Does it work?

**Comment by sounakdey** [comment]
@gnujimmik cuDNN6.0 with cuda 8.0 is givinig the same error for tensorflow 1.1
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
ImportError: libcudnn.so.5: cannot open shared object file: No such file or directory

Please can anyone help.

**Comment by lababidi** [comment]
@sounakdey how did you install tensorflow? you must build it from scratch

**Comment by gnujimmik** [comment]
@sandersk as he mentioned, you will need to build it from source. The default support for the tensorflow is cuda 8.0 and cudnn 5.

**Comment by sounakdey** [comment]
yes i tried to built it from source but it gives an error like this
cuda 8.0.61 cudnn 6
ERROR: /path/to/tensorflow/tensorflow/stream_executor/BUILD:39:1: C++ compilation of rule '//tensorflow/stream_executor:cuda_platform' failed: crosstool_wrapper_driver_is_not_gcc failed: error executing command external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -fPIE -Wall -Wunused-but-set-parameter ... (remaining 130 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
tensorflow/stream_executor/cuda/cuda_blas.cc: In member function 'virtual bool perftools::gputools::cuda::CUDABlas::GetBlasGemmAlgorithms(std::vector<long long int>*)':
tensorflow/stream_executor/cuda/cuda_blas.cc:1916:9: error: 'CUBLAS_GEMM_ALGO5' was not declared in this scope
         CUBLAS_GEMM_ALGO5, CUBLAS_GEMM_ALGO6, CUBLAS_GEMM_ALGO7}) {
         ^
tensorflow/stream_executor/cuda/cuda_blas.cc:1916:28: error: 'CUBLAS_GEMM_ALGO6' was not declared in this scope
         CUBLAS_GEMM_ALGO5, CUBLAS_GEMM_ALGO6, CUBLAS_GEMM_ALGO7}) {
                            ^
tensorflow/stream_executor/cuda/cuda_blas.cc:1916:47: error: 'CUBLAS_GEMM_ALGO7' was not declared in this scope
         CUBLAS_GEMM_ALGO5, CUBLAS_GEMM_ALGO6, CUBLAS_GEMM_ALGO7}) {
any help will be appreciated

**Comment by Androbin** [comment]
@sounakdey This might be #9002

**Comment by sounakdey** [comment]
I saw this but still having the cuda 8.0.61 is giving the same error while building both for r1.1 and r1.2.

**Comment by guschmue** [comment]
@gunan - for windwos cudnn6 is fine, builds and all unit tests are passing. 
All that is needed is to point cmake to the cudnn6 location

```
set CUDNN_HOME=C:\local\cudnn-8.0-windows10-x64-v6.0\cuda
cmake -Dtensorflow_ENABLE_GPU=ON -DCUDNN_HOME=%CUDNN_HOME% 
```

**Comment by gunan** [comment]
Thanks for verification @guschmue !
We are working on installing cudnn on our machines. Once that is done, we will upgrade to cudnn6 accross the board.

**Comment by sounakdey** [comment]
@Mahmoud Yes i built it from scratch but it still never worked while
compiling

**Comment by ernestyalumni** [comment]
2 days ago, virtualenv option install of tensorflow (tf) on Ubuntu 16.04 LTS, CUDNNv6, when import tensorflow, gave error ImportError: libcudnn.so.5: cannot open shared object file: No such file or directory so it was still looking for CUDNN v5, but when I   

```  
pip install --upgrade tensorflow
``` 
(do that with tensorflow-gpu as well) 
then it now works with CUDNNv6 successfully.  So be sure to pip install --upgrade ... (i.e. update pip)  

![tensorflowshortscreenshot from 2017-07-10 03-06-07](https://user-images.githubusercontent.com/3122259/28013512-1b997186-651e-11e7-9357-7dc2b6877ee0.png)

**Comment by dlnp2** [comment]
`cp /usr/lib/x86_64-linux-gnu/libcudnn.so.6 /usr/lib/x86_64-linux-gnu/libcudnn.so.5`  
just worked for me with tensorflow-gpu == 1.2.1. Checked on a bvlc/caffe:gpu docker container with cudnn6 and CUDA8.

**Comment by carlesm** [comment]
I guess that:
`pip install --upgrade tensorflow`
Installs the non-gpu version, so it has no issues with CUDDN 6, logically.

**Comment by sounakdey** [comment]
yes
pip install --upgrade tensorflow-gpu
is still not working with CUDDN 6

**Comment by gunan** [comment]
cudnn v6 support in prebuilt binaries is announced for next release in 1.2 release notes.
Please wait for 1.3 release for it to work out of the box.

**Comment by jubjamie** [comment]
@gunan Should we wait for the full release of 1.3 or should the RC's support v6? Nothing in the rc0 release notes so curious! Ty

**Comment by gunan** [comment]
1.3 RC should be built with cudnn 6.
@av8ramit Could you add cuDNN 6 support to 1.3 release notes?

**Comment by jubjamie** [comment]
@gunan reckon that's related to #11645

**Comment by av8ramit** [comment]
Just FYI, I've added cuDNN 6 to the 1.3 release notes.

**Comment by jubjamie** [comment]
Cheers. Guess this might as well be closed now @cancan101 ?

**Comment by cancan101** [comment]
This can probably be closed though I would ask is cuDNN merely linked against or is TF using the features offered (main ones being dilated convolution and fused conv + bias + activation)?

**Comment by perone** [comment]
It seems that the 1.3.0rc0 is just linked with the cuDNN 6, because I didn't noticed any performance improvement using dilated convolutions with 1.3.0rc0 and cuDNN 6.

**Comment by gunan** [comment]
Correct, we moved the linked library to cudnn 6, but we are still not using the full power of it.
@zheng-xq @tfboyd may have more to say on the timeline for using the new cudnn 6 features.

**Comment by collawolley** [comment]
@gunan @cancan101 @perone HI, I upgraded to tensorflow1.3 and now tensorflow couldn't find my GPU. I reckon it's due to CuDNN5.1. is this assumption true?
How can I upgrade CUDNN5 to CuDNN6 or 7? is it gonna work fine for tensorflow r1.3?
I was only able to download deb file for CuDNN6 and I'm wondering if I have to use the DEB or Pow DEB file. Also, can anybody guide in how to do this upgrade from CuDNN5 to 6 or 7..
Thanks

**Comment by tfboyd** [comment]
@collawolley CUDA and cuDNN are installs that are fairly simple compared to the driver.  The way I do my installed from scratch are as follows and I have a document [here](https://github.com/tfboyd/tf-tools/blob/master/install/aws_ubuntu16_04.md).  The versions of the tools change but the steps are the same.

**GPU Drivers**
I install these with apt-get nvidia-xxx because it is just easier and the drivers that come with the CUDA package are often old and that method seems to be harder.

**CUDA**
Check out my instructions above.  Download the package and you extract it to the folder and ONLY run the CUDA install not the driver install.  My document also covers putting the paths into your ldconfig which I find easier than LD_LIBRARY_PATH although there are advantages to LD_LIBRARY_PATH if you test a lot of different versions of CUDA.  

**cuDNN**
This one might make you laugh.  I do this the cheapest way possible and dump the files in with the CUDA install. Below is the script I use to swap cuDNN versions.  You need to download the cuDNN files first from the NVIDIA site (you have to signup which is 'free') and then you are copying files (maybe 5-10 total) into the CUDA include and lib64 directories.  

You have to change the directories, but this is the general idea and I am sharing it to show that you do not need a package manager and can 100% do this on your own.  That way you "own" it and know exactly what is happening.  

```bash
# Remove current cuDNN.  Commented out for safety uncomment or run manually after finding
# the files
#sudo rm /usr/local/cuda/include/cudnn.h
#sudo rm /usr/local/cuda/lib64/libcudnn*

# Add this version of cuDNN (6.0)
sudo cp -P include/cudnn.h /usr/local/cuda-8.0_cuDNN_7/include/
sudo cp -P lib64/libcudnn* /usr/local/cuda-8.0_cuDNN_7/lib64/
sudo chmod a+r /usr/local/cuda-8.0_cuDNN_7/lib64/libcudnn*
sudo chmod a+r /usr/local/cuda-8.0_cuDNN_7/include/cudnn.h

# assuming you are using the ldconfig approach
sudo ldconfig

```

Good luck.

**Comment by tfboyd** [comment]
Dilated Convolutions support is in progress.  The link below is for me:
b/38417408

Add dilation rate support in Conv2D, Conv3D, and their grad operators.  Here are some of the early numbers for the change.  My guess is this is on a local Titan but possibly something else.  I have a goal to get share information faster and that means data dumps with minimal explanation.  As far as timeline the change is in review and has been in active progress for some time.

```bash
1D convolution benchmark:
input_shape     filter_shape    dilation        padding new_runtime     old_runtime     speedup
===============================================================================
[1,1,2000,1]	[1,3,1,2]	[1,1,2,1]	VALID	0.00005758	0.00005961	1.03526x
[1,1,2000,1]	[1,3,1,2]	[1,1,2,1]	SAME	0.00006395	0.00008794	1.37514x
[1,1,2000,1]	[1,3,1,2]	[1,1,4,1]	VALID	0.00005672	0.00008252	1.45487x
[1,1,2000,1]	[1,3,1,2]	[1,1,4,1]	SAME	0.00005449	0.00008841	1.6225x
[1,1,2000,1]	[1,3,1,2]	[1,1,8,1]	VALID	0.00004836	0.00012469	2.57837x
[1,1,2000,1]	[1,3,1,2]	[1,1,8,1]	SAME	0.00005121	0.00009303	1.81664x
[1,1,2000,1]	[1,3,1,2]	[1,1,16,1]	VALID	0.00006055	0.00008979	1.48291x
[1,1,2000,1]	[1,3,1,2]	[1,1,16,1]	SAME	0.00005921	0.00008144	1.37544x
[1,1,2000,1]	[1,3,1,2]	[1,1,32,1]	VALID	0.00006061	0.00008116	1.33905x
[1,1,2000,1]	[1,3,1,2]	[1,1,32,1]	SAME	0.00005695	0.00009377	1.64653x
[1,1,2000,1]	[1,3,1,2]	[1,1,64,1]	VALID	0.00005906	0.00008995	1.52303x
[1,1,2000,1]	[1,3,1,2]	[1,1,64,1]	SAME	0.00006263	0.00008729	1.39374x
[1,1,2000,1]	[1,3,1,2]	[1,1,128,1]	VALID	0.00006235	0.00008463	1.35734x
[1,1,2000,1]	[1,3,1,2]	[1,1,128,1]	SAME	0.00005383	0.00008037	1.49303x
[1,1,2000,1]	[1,3,1,2]	[1,1,256,1]	VALID	0.00006219	0.00009497	1.52709x
[1,1,2000,1]	[1,3,1,2]	[1,1,256,1]	SAME	0.00006209	0.00009128	1.47012x
[1,1,2000,1]	[1,3,1,2]	[1,1,512,1]	VALID	0.00005178	0.00008769	1.69351x
[1,1,2000,1]	[1,3,1,2]	[1,1,512,1]	SAME	0.00005198	0.00008554	1.64563x
[1,1,2000,4]	[1,3,4,8]	[1,1,2,1]	VALID	0.00006688	0.00011240	1.68062x
[1,1,2000,4]	[1,3,4,8]	[1,1,2,1]	SAME	0.00006801	0.00010370	1.52478x
[1,1,2000,4]	[1,3,4,8]	[1,1,4,1]	VALID	0.00006562	0.00008887	1.35431x
[1,1,2000,4]	[1,3,4,8]	[1,1,4,1]	SAME	0.00005807	0.00009184	1.58154x
[1,1,2000,4]	[1,3,4,8]	[1,1,8,1]	VALID	0.00007196	0.00009882	1.37326x
[1,1,2000,4]	[1,3,4,8]	[1,1,8,1]	SAME	0.00005872	0.00008156	1.38896x
[1,1,2000,4]	[1,3,4,8]	[1,1,16,1]	VALID	0.00005951	0.00011199	1.88187x
[1,1,2000,4]	[1,3,4,8]	[1,1,16,1]	SAME	0.00005866	0.00009291	1.58387x
[1,1,2000,4]	[1,3,4,8]	[1,1,32,1]	VALID	0.00006804	0.00009703	1.42607x
[1,1,2000,4]	[1,3,4,8]	[1,1,32,1]	SAME	0.00007159	0.00008727	1.21903x
[1,1,2000,4]	[1,3,4,8]	[1,1,64,1]	VALID	0.00006002	0.00010006	1.66711x
[1,1,2000,4]	[1,3,4,8]	[1,1,64,1]	SAME	0.00006001	0.00009324	1.55374x
[1,1,2000,4]	[1,3,4,8]	[1,1,128,1]	VALID	0.00006543	0.00010054	1.5366x
[1,1,2000,4]	[1,3,4,8]	[1,1,128,1]	SAME	0.00006640	0.00009464	1.4253x
[1,1,2000,4]	[1,3,4,8]	[1,1,256,1]	VALID	0.00007116	0.00011236	1.57898x
[1,1,2000,4]	[1,3,4,8]	[1,1,256,1]	SAME	0.00006039	0.00012486	2.06756x
[1,1,2000,4]	[1,3,4,8]	[1,1,512,1]	VALID	0.00006599	0.00009332	1.41415x
[1,1,2000,4]	[1,3,4,8]	[1,1,512,1]	SAME	0.00006419	0.00008190	1.2759x
[1,1,2000,16]	[1,3,16,32]	[1,1,2,1]	VALID	0.00006921	0.00009786	1.41396x
[1,1,2000,16]	[1,3,16,32]	[1,1,2,1]	SAME	0.00006978	0.00006958	0.997134x
[1,1,2000,16]	[1,3,16,32]	[1,1,4,1]	VALID	0.00006206	0.00011811	1.90316x
[1,1,2000,16]	[1,3,16,32]	[1,1,4,1]	SAME	0.00006378	0.00010154	1.59204x
[1,1,2000,16]	[1,3,16,32]	[1,1,8,1]	VALID	0.00006083	0.00010898	1.79155x
[1,1,2000,16]	[1,3,16,32]	[1,1,8,1]	SAME	0.00005911	0.00009920	1.67823x
[1,1,2000,16]	[1,3,16,32]	[1,1,16,1]	VALID	0.00005304	0.00009534	1.79751x
[1,1,2000,16]	[1,3,16,32]	[1,1,16,1]	SAME	0.00006025	0.00010284	1.70689x
[1,1,2000,16]	[1,3,16,32]	[1,1,32,1]	VALID	0.00005720	0.00012036	2.1042x
[1,1,2000,16]	[1,3,16,32]	[1,1,32,1]	SAME	0.00006837	0.00011353	1.66052x
[1,1,2000,16]	[1,3,16,32]	[1,1,64,1]	VALID	0.00006880	0.00009820	1.42733x
[1,1,2000,16]	[1,3,16,32]	[1,1,64,1]	SAME	0.00006790	0.00010152	1.49514x
[1,1,2000,16]	[1,3,16,32]	[1,1,128,1]	VALID	0.00006641	0.00011005	1.65713x
[1,1,2000,16]	[1,3,16,32]	[1,1,128,1]	SAME	0.00006811	0.00011749	1.725x
[1,1,2000,16]	[1,3,16,32]	[1,1,256,1]	VALID	0.00006754	0.00011810	1.74859x
[1,1,2000,16]	[1,3,16,32]	[1,1,256,1]	SAME	0.00007047	0.00011071	1.57102x
[1,1,2000,16]	[1,3,16,32]	[1,1,512,1]	VALID	0.00006006	0.00010844	1.80553x
[1,1,2000,16]	[1,3,16,32]	[1,1,512,1]	SAME	0.00006161	0.00009654	1.56695x
[1,1,2000,32]	[1,3,32,64]	[1,1,2,1]	VALID	0.00005919	0.00009413	1.5903x
[1,1,2000,32]	[1,3,32,64]	[1,1,2,1]	SAME	0.00006383	0.00006682	1.04684x
[1,1,2000,32]	[1,3,32,64]	[1,1,4,1]	VALID	0.00006381	0.00009762	1.52985x
[1,1,2000,32]	[1,3,32,64]	[1,1,4,1]	SAME	0.00006399	0.00013848	2.16409x
[1,1,2000,32]	[1,3,32,64]	[1,1,8,1]	VALID	0.00005774	0.00010584	1.83304x
[1,1,2000,32]	[1,3,32,64]	[1,1,8,1]	SAME	0.00007597	0.00008368	1.10149x
[1,1,2000,32]	[1,3,32,64]	[1,1,16,1]	VALID	0.00006298	0.00010941	1.73722x
[1,1,2000,32]	[1,3,32,64]	[1,1,16,1]	SAME	0.00008816	0.00010191	1.15597x
[1,1,2000,32]	[1,3,32,64]	[1,1,32,1]	VALID	0.00007067	0.00009408	1.33126x
[1,1,2000,32]	[1,3,32,64]	[1,1,32,1]	SAME	0.00006937	0.00008838	1.27404x
[1,1,2000,32]	[1,3,32,64]	[1,1,64,1]	VALID	0.00006999	0.00011198	1.59994x
[1,1,2000,32]	[1,3,32,64]	[1,1,64,1]	SAME	0.00006322	0.00009393	1.48576x
[1,1,2000,32]	[1,3,32,64]	[1,1,128,1]	VALID	0.00007443	0.00011252	1.51176x
[1,1,2000,32]	[1,3,32,64]	[1,1,128,1]	SAME	0.00006478	0.00009993	1.54261x
[1,1,2000,32]	[1,3,32,64]	[1,1,256,1]	VALID	0.00008803	0.00011139	1.26536x
[1,1,2000,32]	[1,3,32,64]	[1,1,256,1]	SAME	0.00007771	0.00011038	1.42041x
[1,1,2000,32]	[1,3,32,64]	[1,1,512,1]	VALID	0.00006616	0.00010212	1.54353x
[1,1,2000,32]	[1,3,32,64]	[1,1,512,1]	SAME	0.00005772	0.00011225	1.94473x
[1,1,4000,1]	[1,3,1,2]	[1,1,2,1]	VALID	0.00006751	0.00009870	1.46201x
[1,1,4000,1]	[1,3,1,2]	[1,1,2,1]	SAME	0.00006244	0.00008257	1.32239x
[1,1,4000,1]	[1,3,1,2]	[1,1,4,1]	VALID	0.00006573	0.00010224	1.55545x
[1,1,4000,1]	[1,3,1,2]	[1,1,4,1]	SAME	0.00006055	0.00010280	1.69777x
[1,1,4000,1]	[1,3,1,2]	[1,1,8,1]	VALID	0.00005795	0.00009868	1.70285x
[1,1,4000,1]	[1,3,1,2]	[1,1,8,1]	SAME	0.00005254	0.00008278	1.57556x
[1,1,4000,1]	[1,3,1,2]	[1,1,16,1]	VALID	0.00005476	0.00010069	1.83875x
[1,1,4000,1]	[1,3,1,2]	[1,1,16,1]	SAME	0.00005648	0.00011307	2.00195x
[1,1,4000,1]	[1,3,1,2]	[1,1,32,1]	VALID	0.00006231	0.00007961	1.27764x
[1,1,4000,1]	[1,3,1,2]	[1,1,32,1]	SAME	0.00006199	0.00008299	1.33876x
[1,1,4000,1]	[1,3,1,2]	[1,1,64,1]	VALID	0.00005809	0.00008556	1.47289x
[1,1,4000,1]	[1,3,1,2]	[1,1,64,1]	SAME	0.00005712	0.00008388	1.46849x
[1,1,4000,1]	[1,3,1,2]	[1,1,128,1]	VALID	0.00007150	0.00008308	1.16196x
[1,1,4000,1]	[1,3,1,2]	[1,1,128,1]	SAME	0.00007550	0.00009056	1.19947x
[1,1,4000,1]	[1,3,1,2]	[1,1,256,1]	VALID	0.00007656	0.00008883	1.16027x
[1,1,4000,1]	[1,3,1,2]	[1,1,256,1]	SAME	0.00007246	0.00008632	1.19128x
[1,1,4000,1]	[1,3,1,2]	[1,1,512,1]	VALID	0.00006864	0.00008891	1.29531x
[1,1,4000,1]	[1,3,1,2]	[1,1,512,1]	SAME	0.00006079	0.00008307	1.36651x
[1,1,4000,4]	[1,3,4,8]	[1,1,2,1]	VALID	0.00007227	0.00009067	1.2546x
[1,1,4000,4]	[1,3,4,8]	[1,1,2,1]	SAME	0.00006243	0.00009022	1.44514x
[1,1,4000,4]	[1,3,4,8]	[1,1,4,1]	VALID	0.00006956	0.00012247	1.76064x
[1,1,4000,4]	[1,3,4,8]	[1,1,4,1]	SAME	0.00004862	0.00009418	1.93706x
[1,1,4000,4]	[1,3,4,8]	[1,1,8,1]	VALID	0.00006702	0.00009292	1.38645x
[1,1,4000,4]	[1,3,4,8]	[1,1,8,1]	SAME	0.00006085	0.00009907	1.6281x
[1,1,4000,4]	[1,3,4,8]	[1,1,16,1]	VALID	0.00006899	0.00009074	1.31526x
[1,1,4000,4]	[1,3,4,8]	[1,1,16,1]	SAME	0.00006986	0.00009899	1.41698x
[1,1,4000,4]	[1,3,4,8]	[1,1,32,1]	VALID	0.00006411	0.00009018	1.40664x
[1,1,4000,4]	[1,3,4,8]	[1,1,32,1]	SAME	0.00007178	0.00009497	1.32307x
[1,1,4000,4]	[1,3,4,8]	[1,1,64,1]	VALID	0.00006592	0.00009528	1.44539x
[1,1,4000,4]	[1,3,4,8]	[1,1,64,1]	SAME	0.00006156	0.00009386	1.52469x
[1,1,4000,4]	[1,3,4,8]	[1,1,128,1]	VALID	0.00006765	0.00009771	1.44435x
[1,1,4000,4]	[1,3,4,8]	[1,1,128,1]	SAME	0.00005723	0.00008964	1.56631x
[1,1,4000,4]	[1,3,4,8]	[1,1,256,1]	VALID	0.00007014	0.00009486	1.35244x
[1,1,4000,4]	[1,3,4,8]	[1,1,256,1]	SAME	0.00006361	0.00009410	1.47933x
[1,1,4000,4]	[1,3,4,8]	[1,1,512,1]	VALID	0.00006537	0.00009733	1.48891x
[1,1,4000,4]	[1,3,4,8]	[1,1,512,1]	SAME	0.00007170	0.00010448	1.45718x
[1,1,4000,16]	[1,3,16,32]	[1,1,2,1]	VALID	0.00006054	0.00009855	1.62785x
[1,1,4000,16]	[1,3,16,32]	[1,1,2,1]	SAME	0.00005908	0.00009097	1.53978x
[1,1,4000,16]	[1,3,16,32]	[1,1,4,1]	VALID	0.00006879	0.00009551	1.38843x
[1,1,4000,16]	[1,3,16,32]	[1,1,4,1]	SAME	0.00006928	0.00009561	1.38005x
[1,1,4000,16]	[1,3,16,32]	[1,1,8,1]	VALID	0.00006048	0.00009533	1.57622x
[1,1,4000,16]	[1,3,16,32]	[1,1,8,1]	SAME	0.00006103	0.00009489	1.55481x
[1,1,4000,16]	[1,3,16,32]	[1,1,16,1]	VALID	0.00006214	0.00008976	1.44448x
[1,1,4000,16]	[1,3,16,32]	[1,1,16,1]	SAME	0.00007023	0.00010309	1.46789x
[1,1,4000,16]	[1,3,16,32]	[1,1,32,1]	VALID	0.00006579	0.00009250	1.40599x
[1,1,4000,16]	[1,3,16,32]	[1,1,32,1]	SAME	0.00006850	0.00009546	1.39358x
[1,1,4000,16]	[1,3,16,32]	[1,1,64,1]	VALID	0.00006841	0.00009931	1.45169x
[1,1,4000,16]	[1,3,16,32]	[1,1,64,1]	SAME	0.00007368	0.00009232	1.25299x
[1,1,4000,16]	[1,3,16,32]	[1,1,128,1]	VALID	0.00006102	0.00009517	1.55965x
[1,1,4000,16]	[1,3,16,32]	[1,1,128,1]	SAME	0.00007004	0.00009404	1.34266x
[1,1,4000,16]	[1,3,16,32]	[1,1,256,1]	VALID	0.00005881	0.00009558	1.62523x
[1,1,4000,16]	[1,3,16,32]	[1,1,256,1]	SAME	0.00007120	0.00009117	1.28048x
[1,1,4000,16]	[1,3,16,32]	[1,1,512,1]	VALID	0.00006448	0.00009539	1.47937x
[1,1,4000,16]	[1,3,16,32]	[1,1,512,1]	SAME	0.00005708	0.00009181	1.60844x
[1,1,4000,32]	[1,3,32,64]	[1,1,2,1]	VALID	0.00006209	0.00009588	1.54421x
[1,1,4000,32]	[1,3,32,64]	[1,1,2,1]	SAME	0.00006653	0.00009741	1.46415x
[1,1,4000,32]	[1,3,32,64]	[1,1,4,1]	VALID	0.00006567	0.00009922	1.51089x
[1,1,4000,32]	[1,3,32,64]	[1,1,4,1]	SAME	0.00006993	0.00009407	1.3452x
[1,1,4000,32]	[1,3,32,64]	[1,1,8,1]	VALID	0.00006089	0.00009559	1.56988x
[1,1,4000,32]	[1,3,32,64]	[1,1,8,1]	SAME	0.00006255	0.00010266	1.64125x
[1,1,4000,32]	[1,3,32,64]	[1,1,16,1]	VALID	0.00006349	0.00010176	1.60277x
[1,1,4000,32]	[1,3,32,64]	[1,1,16,1]	SAME	0.00006249	0.00010165	1.62666x
[1,1,4000,32]	[1,3,32,64]	[1,1,32,1]	VALID	0.00006386	0.00010494	1.64328x
[1,1,4000,32]	[1,3,32,64]	[1,1,32,1]	SAME	0.00006792	0.00010760	1.58422x
[1,1,4000,32]	[1,3,32,64]	[1,1,64,1]	VALID	0.00007513	0.00010506	1.39838x
[1,1,4000,32]	[1,3,32,64]	[1,1,64,1]	SAME	0.00006888	0.00008726	1.26684x
[1,1,4000,32]	[1,3,32,64]	[1,1,128,1]	VALID	0.00006989	0.00011145	1.59465x
[1,1,4000,32]	[1,3,32,64]	[1,1,128,1]	SAME	0.00006776	0.00011847	1.74838x
[1,1,4000,32]	[1,3,32,64]	[1,1,256,1]	VALID	0.00007142	0.00011892	1.66508x
[1,1,4000,32]	[1,3,32,64]	[1,1,256,1]	SAME	0.00006216	0.00009617	1.54714x
[1,1,4000,32]	[1,3,32,64]	[1,1,512,1]	VALID	0.00005241	0.00011139	2.12536x
[1,1,4000,32]	[1,3,32,64]	[1,1,512,1]	SAME	0.00006162	0.00009327	1.51363x
[1,1,8000,1]	[1,3,1,2]	[1,1,2,1]	VALID	0.00004589	0.00008756	1.90804x
[1,1,8000,1]	[1,3,1,2]	[1,1,2,1]	SAME	0.00004903	0.00008673	1.76892x
[1,1,8000,1]	[1,3,1,2]	[1,1,4,1]	VALID	0.00005829	0.00008076	1.38549x
[1,1,8000,1]	[1,3,1,2]	[1,1,4,1]	SAME	0.00004904	0.00008209	1.67394x
[1,1,8000,1]	[1,3,1,2]	[1,1,8,1]	VALID	0.00005378	0.00008569	1.59334x
[1,1,8000,1]	[1,3,1,2]	[1,1,8,1]	SAME	0.00006029	0.00009073	1.50489x
[1,1,8000,1]	[1,3,1,2]	[1,1,16,1]	VALID	0.00005733	0.00008718	1.52067x
[1,1,8000,1]	[1,3,1,2]	[1,1,16,1]	SAME	0.00005971	0.00008678	1.45336x
[1,1,8000,1]	[1,3,1,2]	[1,1,32,1]	VALID	0.00005822	0.00008479	1.45637x
[1,1,8000,1]	[1,3,1,2]	[1,1,32,1]	SAME	0.00005659	0.00009635	1.7026x
[1,1,8000,1]	[1,3,1,2]	[1,1,64,1]	VALID	0.00005671	0.00008650	1.5253x
[1,1,8000,1]	[1,3,1,2]	[1,1,64,1]	SAME	0.00006220	0.00011187	1.79855x
[1,1,8000,1]	[1,3,1,2]	[1,1,128,1]	VALID	0.00006788	0.00008434	1.24249x
[1,1,8000,1]	[1,3,1,2]	[1,1,128,1]	SAME	0.00007366	0.00008984	1.21966x
[1,1,8000,1]	[1,3,1,2]	[1,1,256,1]	VALID	0.00006150	0.00008328	1.35415x
[1,1,8000,1]	[1,3,1,2]	[1,1,256,1]	SAME	0.00005964	0.00009823	1.64705x
[1,1,8000,1]	[1,3,1,2]	[1,1,512,1]	VALID	0.00006135	0.00010238	1.66879x
[1,1,8000,1]	[1,3,1,2]	[1,1,512,1]	SAME	0.00005900	0.00008919	1.51169x
[1,1,8000,4]	[1,3,4,8]	[1,1,2,1]	VALID	0.00007488	0.00009541	1.27417x
[1,1,8000,4]	[1,3,4,8]	[1,1,2,1]	SAME	0.00008420	0.00009904	1.17625x
[1,1,8000,4]	[1,3,4,8]	[1,1,4,1]	VALID	0.00007031	0.00010654	1.51529x
[1,1,8000,4]	[1,3,4,8]	[1,1,4,1]	SAME	0.00008041	0.00010912	1.35705x
[1,1,8000,4]	[1,3,4,8]	[1,1,8,1]	VALID	0.00006782	0.00011182	1.64878x
[1,1,8000,4]	[1,3,4,8]	[1,1,8,1]	SAME	0.00008019	0.00011079	1.38159x
[1,1,8000,4]	[1,3,4,8]	[1,1,16,1]	VALID	0.00006540	0.00010130	1.54893x
[1,1,8000,4]	[1,3,4,8]	[1,1,16,1]	SAME	0.00006462	0.00009408	1.4559x
[1,1,8000,4]	[1,3,4,8]	[1,1,32,1]	VALID	0.00006994	0.00010283	1.47026x
[1,1,8000,4]	[1,3,4,8]	[1,1,32,1]	SAME	0.00007176	0.00010054	1.40106x
[1,1,8000,4]	[1,3,4,8]	[1,1,64,1]	VALID	0.00005334	0.00009050	1.69666x
[1,1,8000,4]	[1,3,4,8]	[1,1,64,1]	SAME	0.00007829	0.00009921	1.26721x
[1,1,8000,4]	[1,3,4,8]	[1,1,128,1]	VALID	0.00007156	0.00008490	1.18642x
[1,1,8000,4]	[1,3,4,8]	[1,1,128,1]	SAME	0.00007064	0.00009414	1.33267x
[1,1,8000,4]	[1,3,4,8]	[1,1,256,1]	VALID	0.00008280	0.00010062	1.21522x
[1,1,8000,4]	[1,3,4,8]	[1,1,256,1]	SAME	0.00008573	0.00009233	1.07699x
[1,1,8000,4]	[1,3,4,8]	[1,1,512,1]	VALID	0.00007640	0.00009932	1.3x
[1,1,8000,4]	[1,3,4,8]	[1,1,512,1]	SAME	0.00006977	0.00009816	1.40691x
[1,1,8000,16]	[1,3,16,32]	[1,1,2,1]	VALID	0.00006087	0.00009866	1.62083x
[1,1,8000,16]	[1,3,16,32]	[1,1,2,1]	SAME	0.00005479	0.00009548	1.74265x
[1,1,8000,16]	[1,3,16,32]	[1,1,4,1]	VALID	0.00007926	0.00009564	1.20666x
[1,1,8000,16]	[1,3,16,32]	[1,1,4,1]	SAME	0.00008294	0.00009277	1.11852x
[1,1,8000,16]	[1,3,16,32]	[1,1,8,1]	VALID	0.00007563	0.00009144	1.20904x
[1,1,8000,16]	[1,3,16,32]	[1,1,8,1]	SAME	0.00006585	0.00009198	1.39681x
[1,1,8000,16]	[1,3,16,32]	[1,1,16,1]	VALID	0.00005924	0.00009398	1.58643x
[1,1,8000,16]	[1,3,16,32]	[1,1,16,1]	SAME	0.00007066	0.00009238	1.30739x
[1,1,8000,16]	[1,3,16,32]	[1,1,32,1]	VALID	0.00007358	0.00009095	1.23607x
[1,1,8000,16]	[1,3,16,32]	[1,1,32,1]	SAME	0.00007154	0.00010677	1.49245x
[1,1,8000,16]	[1,3,16,32]	[1,1,64,1]	VALID	0.00008058	0.00008823	1.09494x
[1,1,8000,16]	[1,3,16,32]	[1,1,64,1]	SAME	0.00005409	0.00009674	1.7885x
[1,1,8000,16]	[1,3,16,32]	[1,1,128,1]	VALID	0.00007926	0.00007920	0.999243x
[1,1,8000,16]	[1,3,16,32]	[1,1,128,1]	SAME	0.00006786	0.00008257	1.21677x
[1,1,8000,16]	[1,3,16,32]	[1,1,256,1]	VALID	0.00005607	0.00007784	1.38826x
[1,1,8000,16]	[1,3,16,32]	[1,1,256,1]	SAME	0.00006557	0.00009491	1.44746x
[1,1,8000,16]	[1,3,16,32]	[1,1,512,1]	VALID	0.00006331	0.00009373	1.48049x
[1,1,8000,16]	[1,3,16,32]	[1,1,512,1]	SAME	0.00006038	0.00009216	1.52633x
[1,1,8000,32]	[1,3,32,64]	[1,1,2,1]	VALID	0.00010448	0.00011861	1.13524x
[1,1,8000,32]	[1,3,32,64]	[1,1,2,1]	SAME	0.00010089	0.00011988	1.18822x
[1,1,8000,32]	[1,3,32,64]	[1,1,4,1]	VALID	0.00010399	0.00011929	1.14713x
[1,1,8000,32]	[1,3,32,64]	[1,1,4,1]	SAME	0.00010054	0.00011913	1.1849x
[1,1,8000,32]	[1,3,32,64]	[1,1,8,1]	VALID	0.00010199	0.00011886	1.16541x
[1,1,8000,32]	[1,3,32,64]	[1,1,8,1]	SAME	0.00010086	0.00011846	1.1745x
[1,1,8000,32]	[1,3,32,64]	[1,1,16,1]	VALID	0.00010123	0.00011891	1.17465x
[1,1,8000,32]	[1,3,32,64]	[1,1,16,1]	SAME	0.00010026	0.00011828	1.17973x
[1,1,8000,32]	[1,3,32,64]	[1,1,32,1]	VALID	0.00010027	0.00011974	1.19418x
[1,1,8000,32]	[1,3,32,64]	[1,1,32,1]	SAME	0.00010076	0.00010936	1.08535x
[1,1,8000,32]	[1,3,32,64]	[1,1,64,1]	VALID	0.00009921	0.00010883	1.09697x
[1,1,8000,32]	[1,3,32,64]	[1,1,64,1]	SAME	0.00010124	0.00011203	1.10658x
[1,1,8000,32]	[1,3,32,64]	[1,1,128,1]	VALID	0.00009819	0.00011696	1.19116x
[1,1,8000,32]	[1,3,32,64]	[1,1,128,1]	SAME	0.00010087	0.00012151	1.20462x
[1,1,8000,32]	[1,3,32,64]	[1,1,256,1]	VALID	0.00009623	0.00011365	1.18102x
[1,1,8000,32]	[1,3,32,64]	[1,1,256,1]	SAME	0.00010067	0.00012391	1.23085x
[1,1,8000,32]	[1,3,32,64]	[1,1,512,1]	VALID	0.00009502	0.00012072	1.27047x
[1,1,8000,32]	[1,3,32,64]	[1,1,512,1]	SAME	0.00009995	0.00012979	1.29855x

2D convolution benchmark:
input_shape        filter     dilation  padding new(s)     old(s)      delta
===============================================================================
[1,2000,2000,1]    [3,3,1,2]   [2.2]     VALID 0.00231346  0.00303571  23.7918%
[1,2000,2000,1]    [3,3,1,2]   [2.2]     SAME  0.00228869  0.00302117  24.2449%
[1,2000,2000,1]    [3,3,1,2]   [4,4]     VALID 0.00226216  0.00378979  40.3091%
[1,2000,2000,1]    [3,3,1,2]   [4,4]     SAME  0.00226940  0.00383298  40.7928%
[1,2000,2000,1]    [3,3,1,2]   [8,8]     VALID 0.00224240  0.00380246  41.0277%
[1,2000,2000,1]    [3,3,1,2]   [8,8]     SAME  0.00227775  0.00385336  40.8892%
[1,2000,2000,1]    [3,3,1,2]   [16,16]   VALID 0.00220352  0.00352770  37.5366%
[1,2000,2000,1]    [3,3,1,2]   [16,16]   SAME  0.00227510  0.00367454  38.0848%
[1,2000,2000,1]    [3,3,1,2]   [32,32]   VALID 0.00213600  0.00262561  18.6475%
[1,2000,2000,1]    [3,3,1,2]   [32,32]   SAME  0.00227678  0.00282214  19.3243%
[1,2000,2000,1]    [3,3,1,2]   [64,64]   VALID 0.00200663  0.00219924  8.75803%
[1,2000,2000,1]    [3,3,1,2]   [64,64]   SAME  0.00229604  0.00249575  8.002%
[1,2000,2000,1]    [3,3,1,2]   [128,128] VALID 0.00174985  0.00195712  10.5906%
[1,2000,2000,1]    [3,3,1,2]   [128,128] SAME  0.00228872  0.00247906  7.67791%
[1,2000,2000,1]    [3,3,1,2]   [256,256] VALID 0.00128284  0.00171598  25.2416%
[1,2000,2000,1]    [3,3,1,2]   [256,256] SAME  0.00228398  0.00281686  18.9175%
[1,2000,2000,1]    [3,3,1,2]   [512,512] VALID 0.00056654  0.00089643  36.8004%
[1,2000,2000,1]    [3,3,1,2]   [512,512] SAME  0.00227736  0.00332520  31.5121%
[1,2000,2000,4]    [3,3,4,8]   [2.2]     VALID 0.00621930  0.00720891  13.7276%
[1,2000,2000,4]    [3,3,4,8]   [2.2]     SAME  0.00625408  0.00715719  12.6182%
[1,2000,2000,4]    [3,3,4,8]   [4,4]     VALID 0.00621946  0.00749500  17.0185%
[1,2000,2000,4]    [3,3,4,8]   [4,4]     SAME  0.00627438  0.00715426  12.2987%
[1,2000,2000,4]    [3,3,4,8]   [8,8]     VALID 0.00616579  0.00867273  28.906%
[1,2000,2000,4]    [3,3,4,8]   [8,8]     SAME  0.00625862  0.00875359  28.5022%
[1,2000,2000,4]    [3,3,4,8]   [16,16]   VALID 0.00610005  0.00808240  24.5267%
[1,2000,2000,4]    [3,3,4,8]   [16,16]   SAME  0.00627068  0.00836015  24.9932%
[1,2000,2000,4]    [3,3,4,8]   [32,32]   VALID 0.00591817  0.00744971  20.5584%
[1,2000,2000,4]    [3,3,4,8]   [32,32]   SAME  0.00628644  0.00796382  21.0625%
[1,2000,2000,4]    [3,3,4,8]   [64,64]   VALID 0.00559814  0.00646511  13.41%
[1,2000,2000,4]    [3,3,4,8]   [64,64]   SAME  0.00629064  0.00707921  11.1392%
[1,2000,2000,4]    [3,3,4,8]   [128,128] VALID 0.00491748  0.00555902  11.5405%
[1,2000,2000,4]    [3,3,4,8]   [128,128] SAME  0.00628739  0.00662156  5.0467%
[1,2000,2000,4]    [3,3,4,8]   [256,256] VALID 0.00373130  0.00516305  27.7307%
[1,2000,2000,4]    [3,3,4,8]   [256,256] SAME  0.00627354  0.00846297  25.8707%
[1,2000,2000,4]    [3,3,4,8]   [512,512] VALID 0.00193695  0.00419297  53.8048%
[1,2000,2000,4]    [3,3,4,8]   [512,512] SAME  0.00623270  0.00983202  36.6081%
[1,2000,2000,16]   [3,3,16,32] [2.2]     VALID 0.02008603  0.02596919  22.6544%
[1,2000,2000,16]   [3,3,16,32] [2.2]     SAME  0.02028466  0.02592769  21.7645%
[1,2000,2000,16]   [3,3,16,32] [4,4]     VALID 0.02027318  0.02616444  22.5163%
[1,2000,2000,16]   [3,3,16,32] [4,4]     SAME  0.02056549  0.02615826  21.3805%
[1,2000,2000,16]   [3,3,16,32] [8,8]     VALID 0.02030271  0.02589026  21.5817%
[1,2000,2000,16]   [3,3,16,32] [8,8]     SAME  0.02066941  0.02676808  22.7834%
[1,2000,2000,16]   [3,3,16,32] [16,16]   VALID 0.02005850  0.02607086  23.0616%
[1,2000,2000,16]   [3,3,16,32] [16,16]   SAME  0.02066449  0.02675757  22.7714%
[1,2000,2000,16]   [3,3,16,32] [32,32]   VALID 0.01961490  0.02569756  23.6702%
[1,2000,2000,16]   [3,3,16,32] [32,32]   SAME  0.02073576  0.02729048  24.0183%
[1,2000,2000,16]   [3,3,16,32] [64,64]   VALID 0.01856925  0.02516450  26.2085%
[1,2000,2000,16]   [3,3,16,32] [64,64]   SAME  0.02077435  0.02641020  21.3397%
[1,2000,2000,16]   [3,3,16,32] [128,128] VALID 0.01645060  0.02294116  28.2922%
[1,2000,2000,16]   [3,3,16,32] [128,128] SAME  0.02074842  0.02668388  22.2436%
[1,2000,2000,16]   [3,3,16,32] [256,256] VALID 0.01256535  0.01881233  33.2068%
[1,2000,2000,16]   [3,3,16,32] [256,256] SAME  0.02069790  0.03215248  35.6258%
[1,2000,2000,16]   [3,3,16,32] [512,512] VALID 0.00672528  0.01579534  57.4224%
[1,2000,2000,16]   [3,3,16,32] [512,512] SAME  0.02051509  0.04023022  49.0058%
[1,2000,2000,32]   [3,3,32,64] [2.2]     VALID 0.06883289  0.06289514  -9.44071%
[1,2000,2000,32]   [3,3,32,64] [2.2]     SAME  0.06931305  0.06275911  -10.443%
[1,2000,2000,32]   [3,3,32,64] [4,4]     VALID 0.06910991  0.06272004  -10.1879%
[1,2000,2000,32]   [3,3,32,64] [4,4]     SAME  0.06964424  0.06283241  -10.8413%
[1,2000,2000,32]   [3,3,32,64] [8,8]     VALID 0.06883455  0.06198412  -11.0519%
[1,2000,2000,32]   [3,3,32,64] [8,8]     SAME  0.06975165  0.06324264  -10.2921%
[1,2000,2000,32]   [3,3,32,64] [16,16]   VALID 0.06753337  0.06148470  -9.83768%
[1,2000,2000,32]   [3,3,32,64] [16,16]   SAME  0.06958027  0.06326196  -9.98753%
[1,2000,2000,32]   [3,3,32,64] [32,32]   VALID 0.06529795  0.06071445  -7.54927%
[1,2000,2000,32]   [3,3,32,64] [32,32]   SAME  0.06935539  0.06462759  -7.31545%
[1,2000,2000,32]   [3,3,32,64] [64,64]   VALID 0.06140354  0.05971524  -2.82725%
[1,2000,2000,32]   [3,3,32,64] [64,64]   SAME  0.06933038  0.05810705  -19.3149%
[1,2000,2000,32]   [3,3,32,64] [128,128] VALID 0.05387654  0.05367116  -0.382664%
[1,2000,2000,32]   [3,3,32,64] [128,128] SAME  0.06925929  0.05933531  -16.7253%
[1,2000,2000,32]   [3,3,32,64] [256,256] VALID 0.04047116  0.04335156  6.64428%
[1,2000,2000,32]   [3,3,32,64] [256,256] SAME  0.06909944  0.07228471  4.40656%
[1,2000,2000,32]   [3,3,32,64] [512,512] VALID 0.02030520  0.02902289  30.0373%
[1,2000,2000,32]   [3,3,32,64] [512,512] SAME  0.06882550  0.08068930  14.7031%
```

**Comment by DonghunP** [comment]
Hello. Can you guys help me ? 

I have an error when I do 'import tensorflow as tf' 

python3.5/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so: undefined symbol: cudnnConvolutionBiasActivationForward

How can I fix it?. Thanks...

**Comment by av8ramit** [comment]
@DonghunP what version of TensorFlow are you using? Maybe you are hitting https://github.com/tensorflow/tensorflow/issues/12326

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by gunan** [comment]
I think this is now obsolete, as we are looking into cudnn 7.

**Comment by perone** [comment]
Wasn't this issue being used to track the status of cudnn dilated convolution support ? See previous comments.

**Comment by tfboyd** [comment]
Very old issue but I wanted to add that dilated convs are now supported.  I was searching for issues for another purpose and found this one.  Most/all of you likely already knew this but if not now you do.  :-)  @perone

**Comment by perone** [comment]
That's **amazing**, I wasn't aware, thanks a lot for letting us know @tfboyd.
